src/compass/utils/stitching/stitch_burst.py

```diff
@@ -0,0 +1,404 @@
+                             'among all dates will be stitched (default: None')
+    parser.add_argument('-m', '--margin', type=float,
+                        default=100, dest='margin',
+                        help='Margin to apply during stitching. Same units as bursts coordinate system.'
+                             '(default: 100 m, UTM)')
+    parser.add_argument('-s', '--scratchdir', type=str, default='scratch',
+                        dest='scratch',
+                        help='Directory where to store temporary results (default: scratch)')
+    parser.add_argument('-o', '--outdir', type=str, default='outdir',
+                        dest='outdir',
+                        help='Directory path where to store stitched bursts (default: outdir)')
+    return parser.parse_args()
+
+
+def main(indir, outdir, scratchdir, margin, burst_list):
+    '''
+    Stitch S1-A/B bursts for stack processing
+
+    Parameters:
+    ----------
+    indir: str
+       File path to directory containing S1-A/B bursts
+       organized by date
+    outdir: str
+       File path to directory where to store stitched bursts
+    scratchdir: str
+       File path to directory where to store intermediate
+       results (e.g., shapefiles of burst boundary)
+    margin: float
+       Margin to apply to burst boundary while stitching.
+       Same units as bursts coordinate system
+    burst_list: list [str]
+       List of burst IDs to stitch. If not provided, common
+       bursts among all dates are identified and considered
+       for stitching
+    '''
+    info_channel = journal.info('stitch_burst.main')
+    error_channel = journal.error('stitch_burst.main')
+
+    # Start tracking time
+    info_channel.log('Start burst stitching')
+    t_start = time.time()
+
+    # Check that input directory is valid
+    if not os.path.isdir(indir):
+        err_str = f'{indir} is not a valid input directory'
+        error_channel.log(err_str)
+        raise ValueError(err_str)
+
+    # Create output and scratch dirs if not existing
+    if not os.path.exists(outdir):
+        os.makedirs(outdir, exist_ok=True)
+        helpers.check_write_dir(outdir)
+    if not os.path.exists(scratchdir):
+        os.makedirs(scratchdir, exist_ok=True)
+        helpers.check_write_dir(scratchdir)
+
+    # Collect info for stitching in all dirs in 'indir'
+    # and return a panda dataframe with info
+    data_dict = get_stitching_dict(indir)
+
+    # If stitching some bursts, prune dataframe to
+    # contains only the burst IDs to stitch
+    if burst_list is not None:
+        data_dict = prune_dataframe(data_dict,
+                                    'burst_id', burst_list)
+
+    # Identify common burst IDs among different dates
+    ids2stitch = get_common_burst_ids(data_dict)
+
+    # Prune dataframe to contain only the IDs to stitch
+    data_dict = prune_dataframe(data_dict,
+                                'burst_id', ids2stitch)
```

Comment:
If `prune_dataframe` above is removed, then `get_common_burst_ids` and `prune_dataframe` can be merged into `prune_uncommon_burst_ids` function? Unless you see `prune_dataframe` being being something to be imported and used elsewhere...

Something like:
```python
def prune_uncommon_burst_ids(data):
    '''
    Prune dataframe based on column ID and list of value
    Parameters:
    ----------
    data: pandas.DataFrame
       dataframe that needs to be pruned

    Returns:
    -------
    data: pandas.DataFrame
       Pruned dataframe with rows in 'id_list'
    '''
    unique_dates = list(set(data['date']))

    # Initialize list of unique burst IDs
    common_id = data.burst_id[data.date == unique_dates[0]]

    for date in unique_dates:
        ids = data.burst_id[data.date == date]
        common_id = sorted(list(set(ids.tolist()) & set(common_id)))

    # remove burst IDs not common to all days
    pattern = '|'.join(common_id)
    dataf = data.loc[data['burst_id'].str.contains(pattern,
                                               case=False)]
    return dataf
```

Reply:
Thank you for pointing out commonality with #18; I completely missed this. In light of this, I think `prune_dataframe` and `get_common_burst_ids` could be in their own file in src/compass/utils/dataframe_tools.py as a shared import between stack processing and stitching.